### PR TITLE
Add FdOutputStream

### PR DESCRIPTION
**Pull request by sduzh**
## Problem Summary(Required) ：

Add a new class `FdOutputStream` for writing to a file.

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

### CODE REVIEW

File: be/src/io/fd_output_stream.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    return Status::OK();
+}
+
+Status FdOutputStream::close() {
+    if (_closed) {
+        return Status::OK();
+    }
+    _closed = true;
+    Status st = do_sync_if_needed();
+    int fd = _fd;
```
Comment (wuleistarrocks) [inline be/src/io/fd_output_stream.cpp:92]: no necessary to introduce new variable fd?

Comment (sduzh) [reply be/src/io/fd_output_stream.cpp]: Fixed, PTAL

File: be/src/io/fd_output_stream.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Status FdOutputStream::do_sync_if_needed() {
+    if (_sync_file_on_close) {
+        if (::fdatasync(_fd) != 0) {
+            return io_error(fmt::format("fdatasync({})", _fd), errno);
+        }
+    }
+    if (!_sync_dir.empty()) {
+        int dir_fd;
+        RETRY_ON_EINTR(dir_fd, open(_sync_dir.c_str(), O_DIRECTORY | O_RDONLY));
```
Comment (wyb) [inline be/src/io/fd_output_stream.cpp:74]:
```suggestion
        RETRY_ON_EINTR(dir_fd, ::open(_sync_dir.c_str(), O_DIRECTORY | O_RDONLY));
```

**Review by starrocks-xupeng — DISMISSED** [review]

File: be/src/io/fd_output_stream.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/io/fd_output_stream.cpp]: updated, PTAL

**Review by wyb — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**